### PR TITLE
Add recursive delete of files and folders after installation is finished

### PR DESCRIPTION
**Pull request by Samuell1**
Resolves https://github.com/octobercms/october/issues/4274

### CODE REVIEW

**Comment by w20k** [comment]
Related: https://github.com/octobercms/october/issues/4274

File: install_files/php/Installer.php [hunk 1]
```diff
@@ -873,4 +877,16 @@ protected function validateSqliteFile($filename)
...
         new PDO('sqlite:'.$filename);
     }
+
+    protected function recursiveRemove($dir)
+    {
+        $structure = glob(rtrim($dir, '/').'/*');
+        if (is_array($structure)) {
+            foreach($structure as $file) {
+                if (is_dir($file)) $this->recursiveRemove($file);
+                elseif (is_file($file)) @unlink($file);
```
Comment (LukeTowers) [inline install_files/php/Installer.php]: Please use braces on these conditionals

File: install_files/php/Installer.php [hunk 2]
```diff
@@ -873,4 +877,16 @@ protected function validateSqliteFile($filename)
 
         new PDO('sqlite:'.$filename);
     }
+
+    protected function recursiveRemove($dir)
+    {
+        $structure = glob(rtrim($dir, '/').'/*');
+        if (is_array($structure)) {
+            foreach($structure as $file) {
```
Comment (LukeTowers) [inline install_files/php/Installer.php]: space after foreach please

**Comment by w20k** [comment]
@LukeTowers, sure! Will run do, tomorrow!

**Comment by w20k** [comment]
@LukeTowers works fine, but one minor fix should be added.

It doesn't remove `install.php` (README.md as well)  file from the root directory. And if something failed during the process it adds .log file inside of install_files dir:  `install_files/install.log`. So, when the installation is finished, and successfully, it will still fail. 

Screenshot:
Screenshot 2019-07-29 at 19 42 55

**Comment by LukeTowers** [comment]
@Samuell1 could you look into that please?

**Comment by Samuell1** [comment]
@LukeTowers i think we should move install.log from install_files folder to root.

**Comment by w20k** [comment]
@Samuell1 tried your latest fix. Now install.log is there, install.php was removed, but `dirs` still there.
Screenshot 2019-08-06 at 16 58 20

**Comment by LukeTowers** [comment]
Under what conditions would you retain install.log but not retain the install files? AFAIK install.log is just for if there was an error preventing the installation from completing, in which case you would not want to remove the install_files?

**Comment by Samuell1** [comment]
@LukeTowers install.log is not created only on errors but after every step.

**Comment by LukeTowers** [comment]
@Samuell1 so if there are any errors period the installation will not complete and the files will not be removed? If that's the case then I would prefer that a fully clean install.log (i.e. no errors) just gets removed along with the rest of the installation files.

**Comment by Samuell1** [comment]
I think it should work like that now.

**Comment by LukeTowers** [comment]
@Samuell1 So you can verify that after the install process is successfully completed all install files (install_files directory, install.log, install.php) are completely removed?

**Comment by w20k** [comment]
@Samuell1 ping me if re-test is needed.

**Comment by Samuell1** [comment]
@w20k You can if you have time but i tested it yesterday and works great. I sended a video of steps to @LukeTowers privately.

**Comment by LukeTowers** [comment]
@w20k can you test what happens if the install fails at any point before or during the cleanInstall step? I would like to get rid of the notice at the end telling people to clean up after the installer but I don't want to remove it outright if we aren't handling failure cases properly (such as error during install or incorrect permissions to remove the files required to clean up after the installation process)

**Comment by w20k** [comment]
Sure @LukeTowers will do!

**Comment by Samuell1** [comment]
@LukeTowers if cleaning of files will throw error it will silently continue.

**Comment by ayumi-cloud** [comment]
@Samuell1 please can you watch this video and give us your thoughts: https://www.youtube.com/watch?v=QNJuEWQnBKs&feature=youtu.be

**Pull request merged**